showkase-processor-testing/src/androidTest/java/com/airbnb/android/showkase_processor_testing/ShowkaseBrowserTest.kt

```diff
@@ -0,0 +1,392 @@
+
+        // Select components
+        clickRowWithText("Components")
+
+        // Select Group 3
+        clickRowWithText("Group3")
+
+        // Verify the right composables are visible on the screen
+        verifyRowsWithTextAreDisplayed(listOf("Test Composable4", "Test Composable5"))
+        
+        // Tap on the search icon
+        clickRowWithTag("SearchIcon")
+
+        // Enter "Composable4" in the search field
+        inputTextWithTag("SearchTextField", "Composable4")
+
+        // Ensure that only Composable4 is visible on the screen. The rest of the groups should not be 
+        // visble anymore
+        verifyRowsWithTextAreDisplayed(listOf("Test Composable4"))
+        verifyRowsWithTextDoesNotExist(listOf("Test Composable5"))
+    }
+
+    @Test
+    fun entering_text_in_search_bar_filters_the_visible_colors() {
+        // Ensure all the categories are visible
+        verifyLandingScreen()
+
+        // Select "Colors"
+        clickRowWithText("Colors")
+
+        // Click on "Light Colors"
+        clickRowWithText("Light Colors")
+
+        // Verify the right colors are visible on the screen
+        verifyColorsDetailScreen()
+
+        // Tap on the search icon
+        clickRowWithTag("SearchIcon")
+
+        // Enter "Prim" in the search field
+        inputTextWithTag("SearchTextField", "Prim")
+
+        // Ensure that only "Primary" & "Primary Variant" is visible on the screen. The rest of the 
+        // groups should not be visble anymore
+        verifyRowsWithTextAreDisplayed(listOf("Primary Variant", "Primary"))
+        verifyRowsWithTextDoesNotExist(listOf("Secondary", "Secondary Variant"))
+    }
+
+    @Test
+    fun entering_text_in_search_bar_filters_the_visible_typography() {
+        // Ensure all the categories are visible
+        verifyLandingScreen()
+
+        // Select "Typography"
+        clickRowWithText("Typography")
+
+        // Select Material
+        clickRowWithText("Material")
+
+        // Verify the right typography are visible on the screen
+        verifyTypographyDetailScreen()
+
+        // Swipe back to the top of the sreen
+        onNodeWithTag("TypographyInAGroupList").performGesture {
+            swipeDown()
+        }
+
+        // Tap on the search icon
+        clickRowWithTag("SearchIcon")
+
+        // Enter "Bod" in the search field
+        inputTextWithTag("SearchTextField", "Bod")
+        
+        
+        // Ensure that only "Body1" & "Body2" is visible on the screen. The rest of the groups should
+        // not be visble anymore
+        verifyRowsWithTextAreDisplayed(listOf("Body1", "Body2"))
+        verifyRowsWithTextDoesNotExist(listOf("H1", "H2", "H3", "H4", "H5", "H6", "Subtitle1", 
+            "Subtitle2"))
+    }
+
+    @Test
+    fun navigating_to_component_leaf_screen_and_back_works_ok() {
+        // Ensure all the categories are visible
+        verifyLandingScreen()
+
+        // Select components to go to the component groups screen
```

Comment:
while all these comments overall make the tests easy to understand, they are often redundant as the function names are very readable. redundant comments have the downside of 1. making twice as much text to read to understand the code 2. adding extra maintenance overhead and chance of comments getting out of sync with code

Generally I would recommend only adding comments that add additional information not available from the function invocation (although I wouldn't necessarily recommend changing all these as they are already done and it is quite nice overall)

Reply:
I'll keep this in mind for my future PR's 👍